cpu/samd21/cpu.c

```diff
@@ -31,12 +35,27 @@ static void clk_init(void)
     PM->APBAMASK.reg = (PM_APBAMASK_PM | PM_APBAMASK_SYSCTRL |
                         PM_APBAMASK_GCLK);
 
-    /* adjust NVM wait states, see table 42.30 (p. 1070) in the datasheet */
-#if (CLOCK_CORECLOCK > 24000000)
+    /* adjust NVM wait states, see SAMD21 datasheet
+       Rev A (2017) table 37-40 , page 816 */
     PM->APBBMASK.reg |= PM_APBBMASK_NVMCTRL;
+#if (VDD_MILLIVOLTS > 2700)
+#if (CLOCK_CORECLOCK > 24000000)
     NVMCTRL->CTRLB.reg |= NVMCTRL_CTRLB_RWS(1);
-    PM->APBBMASK.reg &= ~PM_APBBMASK_NVMCTRL;
+#else
+    NVMCTRL->CTRLB.reg |= NVMCTRL_CTRLB_RWS(0);
+#endif
+#else /* VDD_MILLIVOLTS <= 2700 */
+#if (CLOCK_CORECLOCK > 42000000)
+    NVMCTRL->CTRLB.reg |= NVMCTRL_CTRLB_RWS(3);
+#elif (CLOCK_CORECLOCK > 28000000)
+    NVMCTRL->CTRLB.reg |= NVMCTRL_CTRLB_RWS(2);
+#elif (CLOCK_CORECLOCK > 14000000)
```

Comment:
These are getting kind of difficult to read, I wonder if it'd improve readability by using `*_MHz` defines.